src/session.rs

```diff
@@ -296,14 +296,15 @@ impl<C> Session<C>
 where
     C: connect::Connect + Unpin + 'static + Clone + Send + Sync + NewConnector,
 {
-    pub(crate) async fn with_capabilities_and_connector(
+
+    pub(crate) async fn with_capabilities(
```

Comment:
This should just call `with_capabilities_and_connector` with `C::new`.

Reply:
Thinking some more about this, I think `Session` should only have a `with_capabilities_and_connector` constructor. The `Client` should just always be the one to create the connector and pass it in.

Reply:
So we drop `with_capabilities` from `Session`, `new` and `with_capabilities` from generic `Client` and just do something like this?

```rust
#[cfg(feature = "rustls-tls")]
impl RustlsClient {
    pub async fn new_rustls(webdriver: &str) -> Result<Self, error::NewSessionError> {
        Client::new_with_connector(webdriver,hyper_rustls::HttpsConnector::new()).await
    }
   
    pub async fn rustls_with_capabilities(webdriver: &str, cap: webdriver::capabilities::Capabilities) -> Result<Self, error::NewSessionError> {
        Client::with_capabilities_and_connector(webdriver, cap, hyper_rustls::HttpsConnector::new()).await
    }
}
```
And another one for OpenSSL?
This way we can remove the `NewConnector` trait and just have seperate impls for Rustls and Openssl and the user can do just:
`Client::new_rustls("http://localhost:4444").await?`

Reply:
Exactly! Although I would still keep `with_capabilities_and_connector` as a public function on `Client<C>` for anyone who wants to use their own connector type.